pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/OsteoSomaticAssentV3.java

```diff
@@ -0,0 +1,203 @@
+                    "CONSENT_ADDENDUM_PEDIATRIC", STUDY);
+            revisionVariable(templateVariable, meta, version2, variableId);
+        }
+    }
+
+    private void revisionVariableTranslation(TemplateVariable templateVariable,
+                                             RevisionMetadata meta, ActivityVersionDto version3) {
+        log.info("revisioning and updating template variable: {}", templateVariable.getName());
+        Long templateVariableId = sqlHelper.findVariableIdByNameAndActivityId(templateVariable.getName(), version3.getActivityId());
+        log.info("Tmpl variableId  {} ", templateVariableId);
+        revisionVariable(templateVariable, meta, version3, templateVariableId);
+    }
+
+    private void revisionVariable(TemplateVariable templateVariable, RevisionMetadata meta,
+                                  ActivityVersionDto version3, Long templateVariableId) {
+        List<Translation> transList = jdbiVarSubst.fetchSubstitutionsForTemplateVariable(templateVariableId);
+        //all translations have same revision.. hence using first.. lms has only en any way.
+        long revId = jdbiRevision.copyAndTerminate(transList.get(0).getRevisionId().get(), meta);
+        long[] revIds = new long[transList.size()];
+        Arrays.fill(revIds, revId);
+        List<Long> substitutionIds = new ArrayList<>();
+        log.info("subs/translations count: {} for varId : {} ", transList.size(), templateVariableId);
+        for (var currTranslation : transList) {
+            String language = currTranslation.getLanguageCode();
+            String currentText = currTranslation.getText();
+
+            String newTxt = templateVariable.getTranslation(language).get().getText();
+            log.info("terminated: language: {} revId: {} new Text: {} currText: {}", language, currTranslation.getRevisionId(),
+                    newTxt, currentText);
+
+            substitutionIds.add(currTranslation.getId().get());
+            jdbiVarSubst.insert(currTranslation.getLanguageCode(), templateVariable.getTranslation(language).get().getText(),
+                    version3.getRevId(), templateVariableId);
+        }
+        int[] ids = jdbiVarSubst.bulkUpdateRevisionIdsBySubIds(substitutionIds, revIds);
+        if (ids.length != revIds.length) {
+            throw new DDPException("returned ids length " + ids.length + "  doesnt match revIds passed length " + revIds.length);
+        }
+        log.info("revisioned and updated template variable: {}", templateVariableId);
+    }
+
+    private interface SqlHelper extends SqlObject {
+
+
+        /* Find the content block that has the given body template text. Make sure it is from a block that belongs in the expected activity
+         * (and thus the expected study). This is done using a `union` subquery to find all the top-level and nested block ids for the
+         * activity and using that to match on the content block.
+         */
```

Comment:
I like this. Are variable names not unique in a study? Yet, this still provides a good protection in case of accidental duplicated variable names.